packages/react-instantsearch-core-next/src/__tests__/useConnector.test.tsx

```diff
@@ -0,0 +1,70 @@
+import { renderHook } from '@testing-library/react-hooks';
+import React from 'react';
+
+import { createSearchClient } from '../../../../test/mock';
+import { InstantSearch } from '../InstantSearch';
+import { useConnector } from '../useConnector';
+import { noop } from '../utils';
+
+import type { Connector } from 'instantsearch.js';
+
+type CustomSearchBoxConnector = {
+  $$type: 'test.searchBox';
+  renderState: {
+    query: string;
+  };
+};
+
+const connectCustomSearchBox: Connector<
+  CustomSearchBoxConnector,
+  Record<string, never>
+> =
+  (renderFn, unmountFn = noop) =>
+  (widgetParams) => ({
+    $$type: 'test.searchBox',
+    init({ instantSearchInstance }) {
+      renderFn(
+        { query: 'query at init', instantSearchInstance, widgetParams },
+        true
+      );
+    },
+    render({ instantSearchInstance }) {
+      renderFn({ query: 'query', instantSearchInstance, widgetParams }, false);
+    },
+    dispose() {
+      unmountFn();
+    },
+  });
+
+describe('useConnector', () => {
+  test('returns the connector render state', async () => {
```

Comment:
we should also test that the widget has been unmounted and mounted to the parent (the right parent) or do you consider that testing React?

Reply:
Yes, that's an important part indeed. Added a test in 87c3f0215ef0bc250098413de34a02aaa1c92301.